src/main/java/edu/ie3/datamodel/io/factory/result/SystemParticipantResultFactory.java

```diff
@@ -57,7 +57,7 @@ public SystemParticipantResultFactory() {
 
     if (data.getTargetClass().equals(HpResult.class)
         || data.getTargetClass().equals(ChpResult.class)) {
```

Comment:
Additional remark: You can simplify this for future cases, where classes extend `SystemParticipantWithHeatResult`:

```suggestion
    if (SystemParticipantWithHeatResult.class.isAssignableFrom(data.getTargetClass()) {
```

Reply:
Sure, I will make the change.